cert/selfsigned.go

```diff
@@ -100,6 +100,52 @@ func dnsNames(tlsExtraDomains []string) (string, []string) {
 	return host, dnsNames
 }
 
+// Outdated returns whether the given certificate is outdated wrt the IPs and
+// domains.
+func Outdated(cert *x509.Certificate, tlsExtraIPs,
+	tlsExtraDomains []string) (bool, error) {
+
+	// Parse the slice of IP strings.
+	ips, err := ipAddresses(tlsExtraIPs)
+	if err != nil {
+		return false, err
+	}
+
+	// If the certificate has a different number of IP addresses, it is
+	// definitely out of date.
+	if len(ips) != len(cert.IPAddresses) {
```

Comment:
Alternatively, should we relax this to allow _more_ IPs in the cert than are specified in our config? A sample case being a cert that a user has generated themselves, and synced with the existing configuration options.

Reply:
The problem then is that even though the user removes an IP from the config, the cert will still allow it.

Reply:
this will return a false positive if the user accidentally includes a duplicate IP, when in fact we don't need to change the cert. imo it'd be better to convert both to sets, then compare length.

one possible downside of this is that certs generated before this may not have checked for duplicates, so it would cause those to be refreshed. perhaps we should start checking for duplicate entries before creating new certs as well?

Reply:
Yep, fixed: https://github.com/lightningnetwork/lnd/pull/3932/commits/7aba914bbe87456f039f245afc54b771baa398c8